### PR TITLE
Fix nightly, backport and merge-queue

### PR DESCRIPTION
**Pull request by GuyAv46**
**Describe the changes in the pull request**

Fixed unit-test flow for OSs that don't support node20
Added slack notification on nightly failure
Update backport flow (aligned with RediSearch)

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: .github/workflows/task-unit-test.yml [hunk 1]
```diff
@@ -30,7 +30,24 @@ jobs:
       - name: pre-checkout script
         if: ${{ inputs.pre-checkout-script }}
         run: ${{ inputs.pre-checkout-script }}
+      - name: Check for node20 support
+        id: node20
+        run: |
+          for os in amazonlinux:2 ubuntu:bionic; do
```
Comment (meiravgri) [inline .github/workflows/task-unit-test.yml:36]: is there a more general way to test support?
something similar to `which node20`?

File: .github/workflows/task-unit-test.yml [hunk 2]
```diff
@@ -40,8 +57,14 @@ jobs:
...
-        if: ${{ inputs.run-valgrind }} && failure()
-        uses: actions/upload-artifact@v3
+        if: ${{ inputs.run-valgrind }} && failure() && steps.node20.outputs.supported == 'true'
+        uses: actions/upload-artifact@v4
+        with:
+          name: valgrind tests reports
+          path: bin/Linux-x86_64-debug/unit_tests/Testing/Temporary/
+      - name: Archive valgrind tests reports (fallback)
+        if: ${{ inputs.run-valgrind }} && failure() && steps.node20.outputs.supported == 'false'
+        uses: actions/upload-artifact@v4
```
Comment (meiravgri) [inline .github/workflows/task-unit-test.yml:67]: 3?

File: .github/workflows/task-unit-test.yml (hunk 1 above)
Comment (GuyAv46) [reply .github/workflows/task-unit-test.yml:36]: I think this is the simplest way (as we do in RediSearch). The node20 comes from the runner but fails in runtime due to a missing and unsupported glibc version, so I don't think `which node20` will fail/return nothing

File: .github/workflows/task-unit-test.yml (hunk 2 above)
Comment (meiravgri) [reply .github/workflows/task-unit-test.yml]: maybe trigger the temp flow?

File: .github/workflows/task-unit-test.yml (hunk 1 above)
Comment (GuyAv46) [reply .github/workflows/task-unit-test.yml:36]: This is a temporary solution regardless. We hope to drop the support for these OSs soon. When we do, I hope it is clear that this step and all the ones that need its conclusion should be removed

**Review by meiravgri — DISMISSED** [review]

**Review by meiravgri — APPROVED** [review]

**Pull request merged**